File: Native/Test/Python/dependency-loop.py
import fabric
fabricClient = fabric.createClient()

node1 = fabricClient.DependencyGraph.createNode("one")
node2 = fabricClient.DependencyGraph.createNode("two")

try:
  node1.setDependency( node1, "reflex" )
  fabricClient.flush()
except Exception as e:
  print e

try:
  node1.setDependency( node2, "one" )
  node2.setDependency( node1, "two" )
  fabricClient.flush()
except Exception as e:
  print e

fabricClient.close()

File: Native/Test/Node/dependency-loop.js
/*
 *  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
 */

FABRIC = require('Fabric').createClient();

node1 = FABRIC.DependencyGraph.createNode("one");
node2 = FABRIC.DependencyGraph.createNode("two");

try {
  node1.setDependency( node1, "reflex" );
  FABRIC.flush();
}
catch (e) {
  console.log( e );
}

try {
  node1.setDependency( node2, "one" );
  node2.setDependency( node1, "two" );
  FABRIC.flush();
}
catch (e) {
  console.log( e );
}

FABRIC.close();
